config/src/test/java/org/springframework/security/config/http/OAuth2LoginBeanDefinitionParserTests.java

```diff
@@ -0,0 +1,739 @@
+			throws Exception {
+		this.spring.configLocations(this.xml("SingleClientRegistration-WithCustomAuthorizationRequestResolver"))
+				.autowire();
+
+		this.mvc.perform(get("/oauth2/authorization/google")).andExpect(status().is3xxRedirection())
+				.andExpect(redirectedUrl(MockAuthorizationRequestResolver.REQUEST_URI));
+	}
+
+	// gh-5347
+	@Test
+	public void requestWhenMultipleClientsConfiguredThenRedirectDefaultLoginPage() throws Exception {
+		this.spring.configLocations(this.xml("MultiClientRegistration")).autowire();
+
+		this.mvc.perform(get("/")).andExpect(status().is3xxRedirection())
+				.andExpect(redirectedUrl("http://localhost/login"));
+	}
+
+	@Test
+	public void successLoginWhenSingleClientRegistrationAndCustomLoginPageThenRedirectToCustomLoginPage()
+			throws Exception {
+		this.spring.configLocations(this.xml("SingleClientRegistration-WithCustomLoginPage")).autowire();
+
+		this.mvc.perform(get("/")).andExpect(status().is3xxRedirection())
+				.andExpect(redirectedUrl("http://localhost/custom-login"));
+	}
+
+	// gh-6802
+	@Test
+	public void requestWhenSingleClientRegistrationWithFormLoginPageThenRedirectDefaultLoginPage() throws Exception {
+		this.spring.configLocations(this.xml("SingleClientRegistration-WithFormLogin")).autowire();
+
+		this.mvc.perform(get("/")).andExpect(status().is3xxRedirection())
+				.andExpect(redirectedUrl("http://localhost/login"));
+	}
+
+	@Test
+	public void customClientRegistrationRepository() throws Exception {
+		this.spring.configLocations(this.xml("WithCustomClientRegistrationRepository")).autowire();
+
+		assertThat(clientRegistrationRepository).isInstanceOf(MockClientRegistrationRepository.class);
+		ClientRegistration clientReg = clientRegistrationRepository.findByRegistrationId("mock");
+		assertThat(clientReg.getRegistrationId()).isEqualTo("mock");
+
+		Field field = oauth2LoginAuthenticationFilter.getClass().getDeclaredField("clientRegistrationRepository");
+		field.setAccessible(true);
+		Object fieldVal = field.get(oauth2LoginAuthenticationFilter);
+		assertThat(fieldVal).isInstanceOf(MockClientRegistrationRepository.class);
+		assertThat(fieldVal).isSameAs(clientRegistrationRepository);
+	}
+
+	@Test
+	public void customAuthorizedClientRepository() throws Exception {
+		this.spring.configLocations(this.xml("WithCustomAuthorizedClientRepository")).autowire();
+
+		assertThat(oauth2AuthorizedClientRepository).isInstanceOf(MockOAuth2AuthorizedClientRepository.class);
+
+		Field field = oauth2LoginAuthenticationFilter.getClass().getDeclaredField("authorizedClientRepository");
+		field.setAccessible(true);
+		Object fieldVal = field.get(oauth2LoginAuthenticationFilter);
+		assertThat(fieldVal).isInstanceOf(MockOAuth2AuthorizedClientRepository.class);
+		assertThat(fieldVal).isSameAs(oauth2AuthorizedClientRepository);
+	}
+
+	@Test
+	public void customAuthorizedClientService() throws Exception {
```

Comment:
Please update test to `verify()` that the mock `OAuth2AuthorizedClientService` is called. We can also rename to `requestWhenCustomAuthorizedClientServiceThenCalled`